client/lib/site/timezone.js

```diff
@@ -20,12 +20,12 @@ import moment, { MomentInput, Moment } from 'moment-timezone';
  *                        Moment with gmtOffset applied if no timezone is provided.
  *                        If neither is provided, the original moment is returned.
  */
-export function applySiteOffset( input, { timezone, gmtOffset } ) {
+export function applySiteOffset( input, { timezone, gmtOffset, keepLocalTime = false } ) {
 	if ( timezone ) {
 		return moment.tz( input, timezone );
```

Comment:
When `input` already has a timezone attached, Moment falls back to its default behavior, which is to parse and update the timezone, effectively ignoring our new parameter. I think I have a way to get around that, though:

```suggestion
		return moment( input ).tz( timezone, keepLocalTime );
```

Reply:
I pushed a new version with this exact change. Thanks, Matt!